packages/atoms/availabilitylist/ScheduleListItem.tsx

```diff
@@ -0,0 +1,134 @@
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+} from "@/components/ui/dropdown-menu";
+import { Toaster } from "@/components/ui/toaster";
+import { useToast } from "@/components/ui/use-toast";
+import { Globe, MoreHorizontal, Star, Copy, Trash } from "lucide-react";
+import { Fragment } from "react";
+
+import { availabilityAsString } from "@calcom/lib/availability";
+
+type Schedule = {
+  isDefault: boolean;
+  id: number;
+  name: string;
+  availability: {
+    id: number;
+    startTime: Date;
+    endTime: Date;
+    userId: number | null;
+    eventTypeId: number | null;
+    date: Date | null;
+    days: number[];
+    scheduleId: number | null;
+  }[];
+  timezone: string | null;
+};
+
+export function Availability({
+  schedule,
+  isDeletable,
+  displayOptions,
+  updateDefault,
+  duplicateFunction,
+  deleteFunction,
+}: {
+  schedule: Schedule;
+  isDeletable: boolean;
+  updateDefault: ({ scheduleId, isDefault }: { scheduleId: number; isDefault: boolean }) => void;
+  duplicateFunction: ({ scheduleId }: { scheduleId: number }) => void;
+  deleteFunction: ({ scheduleId }: { scheduleId: number }) => void;
+  displayOptions?: {
+    timeZone?: string;
+    hour12?: boolean;
+  };
+}) {
+  const { toast } = useToast();
+
+  return (
+    <li key={schedule.id}>
```

Comment:
Can the returned JSX be split into smaller components that represent what we see in the UI, like <TimeZone> and for example 3 dots with Duplicate / Delete under <Controls> component etc ? It is hard to understand what's going on.

Reply:
@supalarry true the dropdown is a bit overwhelming so that can go into a separate `Controls` component but is there really a need to create a `TimeZone` component? I don't think that part contains enough code to go into a separate component for itself